src/metabase/api/app.clj

```diff
@@ -68,4 +76,139 @@
   [id]
   (hydrate-details (api/read-check App id)))
 
+(defn- replace-scaffold-targets
+  [structure scaffold-target->target-id]
+  (walk/postwalk
+    (fn [node]
+      (if (and (vector? node) (= "scaffold-target-id" (first node)))
```

Comment:
I suppose this check is not really necessary, the next line will not match anyway if this does not. If you keep this check, you can make the `scaffold-target->target-id` map simpler by omitting the tag around the id.